ibis/expr/types/joins.py

```diff
@@ -154,7 +249,15 @@ def __init__(self, arg, collisions=None):
             arg = ops.JoinTable(arg, index=0)
             arg = ops.JoinChain(arg, rest=(), values=arg.fields)
         super().__init__(arg)
+        # the collisions and equalities are used to track the name collisions
+        # and the equality groups join fields based on equality predicates;
+        # these must be tracked in the join expression because the join chain
+        # operation doesn't hold any information about `lname` and `rname`
+        # parameters passed to the join methods and used to disambiguate field
+        # names; the collisions are used to raise an error if there are any
+        # name collisions after the join chain is finished
         object.__setattr__(self, "_collisions", collisions or set())
+        object.__setattr__(self, "_equalities", equalities or DisjointSet())
```

Comment:
Just to confirm: there's an instance of `DisjoinSet` per instance of `Join`, and each instance of `Join` corresponds to a chain of joins. Is that correct?

Reply:
Correct. The disjoint set gets copied as new links aded to the join chain.